pkg/cloudprovider/providers/azure/azure_vmss.go

```diff
@@ -435,9 +435,14 @@ func (ss *scaleSet) GetPrimaryInterface(nodeName, vmSetName string) (network.Int
 		return network.Interface{}, err
 	}
 
-	// Check scale set name.
-	// Backends of Standard load balancer could belong to multiple VMSS, so we don't check vmSet for it.
-	if !strings.EqualFold(ssName, vmSetName) && !ss.useStandardLoadBalancer() {
+	// Check scale set name. Note that vmSetName is empty string when getting the
+	// Node's IP address. While vmSetName is not empty, it should be checked with
+	// Node's real scale set name:
+	// - For basic SKU load balancer, errNotInVMSet should be returned if the node's
+	//   scale set is mismatched with vmSetName.
+	// - For standard SKU load balancer, backend could belong to multiple VMSS, so we
+	//   don't check vmSet for it.
+	if vmSetName != "" && !ss.useStandardLoadBalancer() && !strings.EqualFold(ssName, vmSetName) {
```

Comment:
For scaleSet, is there still any place GetPrimaryInterface is called with vmSetName!=null? Can we just remove the vmSetName parameter. GetPrimaryInterface could be private methods which can be removed from 'VMSet interface'?

Reply:
seems not. Let me make it private

Reply:
wait, there is a `availalbilitySet` wrap within scaleSet, which requires it within VMSet interface:

https://github.com/kubernetes/kubernetes/blob/5ba149e182da67e386e45910af6d875603a766f0/pkg/cloudprovider/providers/azure/azure_vmss.go#L427-L429

Reply:
That call also pass "" to vmSetName. Seems the ideal interface here is: GetPrimaryInterface(nodeName string)

The only usage of vmSet in availabilitySet.GetPrimaryInterface is:
https://github.com/kubernetes/kubernetes/blob/b976ebd56a07ebb009aeb8f14913e3472e5609ee/pkg/cloudprovider/providers/azure/azure_standard.go#L578-L585

Can we move this section to 'ensureHostInPool', which is the only place requires vmSet validation. It looks like more the functionality of 'ensureHostInPool'.